src/app/components/shared/datetime/datetime/datetime.component.spec.ts

```diff
@@ -0,0 +1,230 @@
+    expectedDateTimeAttribute: "2020-01-01T20:10:11.000+08:00",
+  },
+  {
+    name: "should display the correct dateTime if the Luxon dateTime object is the same timezone as the users local timezone",
+    // by setting the timezone to Australia/Perth, we should see that the time value is not changed
+    // because the Luxon DateTime object is already in the same timezone as the test runner
+    // by using { keepLocalTime: true }, we ensure that the time is not changed when we change the timezone
+    // without using { keepLocalTime: true }, the DateTime object would be implicitly created as 20:10:11 +08:00
+    value: DateTime.fromISO("2020-01-01T12:10:11.000Z").setZone(
+      "Australia/Perth",
+      { keepLocalTime: true }
+    ),
+    expectedText: "2020-01-01 12:10:11",
+    expectedTooltip: "2020-01-01 12:10:11.000 Australia/Perth +08:00",
+    expectedDateTimeAttribute: "2020-01-01T12:10:11.000+08:00",
+  },
+  {
+    // most of the tests create the DateTime objects in UTC+00:00
+    // for this test, we create th DateTime object in UTC+02:00
+    // we should therefore see the hours increase by 6
+    name: "should display the correct dateTime if the Luxon dateTime object is a different utc offset to the users local timezone",
+    // by using +02:00, we know the offset, but not the timezone
+    // however, because this value will be localized to the users local timezone, we should see the timezone emitted
+    // in the users local timezone
+    value: DateTime.fromISO("2020-01-01T12:10:11.000+02:00"),
+    expectedText: "2020-01-01 18:10:11",
+    // notice that the tooltips are still in the test runners mock timezone
+    // this is because the emitted date/time has been localized to the test runners timezone
+    expectedTooltip: "2020-01-01 18:10:11.000 Australia/Perth +08:00",
+    expectedDateTimeAttribute: "2020-01-01T18:10:11.000+08:00",
+  },
+  {
+    // this test is the same as the previous offset test, except that we know both the timezone and offset
+    // in this test. (as opposed to the previous test where we only knew the offset)
+    name: "should display the correct DateTime if the Luxon dateTime object is a different timezone to the users local timezone",
+    // by using Australia/Sydney, we know the timezone and offset
+    value: DateTime.fromISO("2020-01-01T12:10:11.000Z").setZone(
+      "Australia/Sydney",
+      { keepLocalTime: true }
+    ),
+    expectedText: "2020-01-01 09:10:11",
+    expectedTooltip: "2020-01-01 09:10:11.000 Australia/Perth +08:00",
+    expectedDateTimeAttribute: "2020-01-01T09:10:11.000+08:00",
+  },
+  {
+    name: "should increment date if timezone localization causes the date to increment",
+    value: DateTime.fromISO("2020-01-01T23:10:11.000Z"),
+    expectedText: "2020-01-02 07:10:11",
+    expectedTooltip: "2020-01-02 07:10:11.000 Australia/Perth +08:00",
+    expectedDateTimeAttribute: "2020-01-02T07:10:11.000+08:00",
+  },
+  {
```

Comment:
Do you really need test this? Isn't this basically testing the luxon library, which is already tested?